### PR TITLE
HHH-16144 AbstractManagedType should look up declared concrete generic attributes before super attributes

### PR DESCRIPTION
**Pull request by quaff**
Fix 
```
java.lang.IllegalArgumentException: Unable to locate Attribute with the given name [xxx] on this ManagedType [java.lang.Object]
	at org.hibernate.metamodel.model.domain.AbstractManagedType.checkNotNull(AbstractManagedType.java:225) ~[hibernate-core-6.3.1.Final.jar:6.3.1.Final]
	at org.hibernate.metamodel.model.domain.AbstractManagedType.getAttribute(AbstractManagedType.java:148) ~[hibernate-core-6.3.1.Final.jar:6.3.1.Final]
	at org.hibernate.metamodel.model.domain.AbstractManagedType.getAttribute(AbstractManagedType.java:43) ~[hibernate-core-6.3.1.Final.jar:6.3.1.Final]
	at org.springframework.data.jpa.repository.query.QueryUtils.requiresOuterJoin(QueryUtils.java:836) ~[spring-data-jpa-3.2.0.jar:3.2.0]
```

https://hibernate.atlassian.net/browse/HHH-16144

### CODE REVIEW

**Comment by quaff** [comment]
Could you take a look? @mbladel

**Review by mbladel — COMMENTED** [review]
This change is not correct, and the failing tests you changed confirm that. The reported java type for generic paths should be the generic one. Using generic paths in Hibernate queries works correctly, so the issue described in the Jira is due to problematic code on Spring's side, as we had already said.

File: hibernate-core/src/test/java/org/hibernate/orm/test/metamodel/genericmodel/GenericMappedSuperclassMetamodelTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		}
+)
+public class GenericMappedSuperclassMetamodelTest {
+
+	@Test
+	public void testGenericAttributeFromMappedSuperclassIsNotTypeErased(EntityManagerFactoryScope scope) {
+		scope.inTransaction( entityManager -> {
+			EntityType<Book> bookType = entityManager.getMetamodel().entity( Book.class );
+			assertEquals( Owner.class, bookType.getAttribute("owner").getJavaType() );
+		});
```
Comment (gavinking) [inline hibernate-core/src/test/java/org/hibernate/orm/test/metamodel/genericmodel/GenericMappedSuperclassMetamodelTest.java:37]: I don't think this assertion is correct. As with Java's reflection API, `EntityType.getAttribute()` gives you a representation of the inherited attribute _as it was declared in the supertype_. It does _not_ do substitution of generic type arguments from the inheriting subclass, and I don't see any support anywhere in the JPA API or specification which hints that it should.

So I believe that it's correct for Hibernate to return `Object.class` here, and I think this PR should be closed.

**Review by gavinking — CHANGES_REQUESTED** [review]

**Comment by mbladel** [comment]
Closing this.

**Pull request closed without merging**